lib/manager/selectors/index.js

```diff
@@ -1,11 +1,205 @@
+  (timetables: Timetables, filters: Filters) => {
+    if (!timetables.data.feed) return {columns: [], rows: []}
+
+    // assume 1st pattern is the only one we're interested in
+    const pattern = timetables.data.feed.patterns[0]
+
+    // there could be no patterns found or no trips associate with that pattern
+    // if eiter happens, return blank
+    if (!pattern || pattern.trips.length === 0) return {columns: [], rows: []}
+
+    // create stop lookup because the stops come back out-of-order
+    const stopLookup: { [string]: Stop } = {}
+    const stopHasStopTimeWithTime: { [string]: boolean } = {}
+    pattern.stops.forEach(stop => {
+      stopLookup[stop.stop_id] = stop
+    })
+
+    // begin making column headers with trip info
+    let columns: Array<Column> = [
+      {
+        dataField: 'tripId',
+        isKey: true,
+        name: 'Trip ID',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      },
+      {
+        dataField: 'tripHeadsign',
+        name: 'Trip Headsign',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      },
+      {
+        dataField: 'tripShortName',
+        name: 'Trip Short Name',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      }
+    ]
+
+    // make rest of the row headers by making stop names as headers
+    // assume that the first trip is representative of stops in timetable
+    const firstTrip = pattern.trips[0]
+
+    // sort stop times by stop_sequence just in case
+    firstTrip.stop_times.sort((a, b) => a.stop_sequence - b.stop_sequence)
+    const stopColWidth = 200
+    firstTrip.stop_times.forEach(stopTime => {
+      const { stop_id: stopId } = stopTime
+      columns.push({
+        colSpan: filters.showArrivals ? '2' : '1',
+        dataField: filters.showArrivals ? undefined : `stop-${stopId}-departure`,
+        name: stopLookup[stopId].stop_name,
+        row: '0',
+        width: stopColWidth
+      })
+      if (filters.showArrivals) {
+        columns.push({
+          dataField: `stop-${stopId}-arrival`,
+          name: 'Arrive',
+          row: '1',
+          width: stopColWidth / 2
+        })
+        columns.push({
+          dataField: `stop-${stopId}-departure`,
+          name: 'Depart',
+          row: '1',
+          width: stopColWidth / 2
+        })
+      }
+    })
+
+    // add row data
+    const rows = pattern.trips.map(trip => {
+      const row = {
+        tripId: trip.trip_id,
+        tripHeadsign: trip.trip_headsign,
+        tripShortName: trip.trip_short_name
+      }
+      trip.stop_times.forEach(stopTime => {
+        if (
+          isNumber(stopTime.arrival_time) ||
+          isNumber(stopTime.departure_time)
+        ) {
+          stopHasStopTimeWithTime[stopTime.stop_id] = true
+        }
+        row[`stop-${stopTime.stop_id}-arrival-raw`] = stopTime.arrival_time
```

Comment:
A comment about how these fields are used would be helpful (i.e., why is the raw value preserved?). Edit: ah, probably for sorting below.  Still, a comment would be good.